tests/test_coordinateConverter.py

```diff
@@ -0,0 +1,556 @@
+
+                # beam 3 should on detector D1, 3/4 of the way from LL to UR
+                beamInfo3 = self.cco["3"]
+                self.assertTrue(beamInfo3.isOnDetector)
+                self.assertEqual(beamInfo3.detectorName, "D1")
+                self.assertPairsAlmostEqual(beamInfo3.detectorPos, detector34Pos,
+                                            maxDiff=maxDetectorPosError)
+
+                # beam 4 should land on the center of detector D2
+                beamInfo4 = self.cco["4"]
+                self.assertTrue(beamInfo4.isOnDetector)
+                self.assertEqual(beamInfo4.detectorName, "D2")
+                self.assertPairsAlmostEqual(beamInfo4.detectorPos, detectorCtrPos,
+                                            maxDiff=maxDetectorPosError)
+
+                # beam 5 should on detector D2, 3/4 of the way from LL to UR
+                beamInfo5 = self.cco["5"]
+                self.assertTrue(beamInfo5.isOnDetector)
+                self.assertEqual(beamInfo5.detectorName, "D2")
+                self.assertPairsAlmostEqual(beamInfo5.detectorPos, detector34Pos,
+                                            maxDiff=maxDetectorPosError)
+
+    def testSetPupilFieldAngle(self):
+        for pupilFieldAngle, pupilPos, beam in itertools.product(
+            ((0, 0), (0, 0.05), (-0.05, -0.03)),
+            ((0, 0), (0, 5000), (-5000, 0), (5000, -5000)),
+            self.cco.beamNames,
+        ):
+            with self.subTest(pupilFieldAngle=pupilFieldAngle, pupilPos=pupilPos, beam=beam):
+                self.cco.setPupilFieldAngle(pupilPos=pupilPos, pupilFieldAngle=pupilFieldAngle, beam=beam)
+                beamInfo = self.cco[beam]
+                self.assertPairsAlmostEqual(beamInfo.pupilPos, pupilPos, self.maxPupilPosErr)
+                self.assertPairsAlmostEqual(beamInfo.pupilFieldAngle, pupilFieldAngle, maxDiff=1e-5)
+                self.checkOrientation()
+
+    def testAngleProperties(self):
+        """Test cbpAzAltObserved, telAzAltObserved, telRotObserved
+        and their Internal equivalents
+        """
+        # set non-trivial scale and offset for all axes;
+        # azimuth and rotator scales must be ±1 in order to handle wrap correctly;
+        # altitude scales should be nearly 1 and altitudeoffsets should be small
+        # to avoid hitting limits
+        self.cco.config.cbpAzAltScale = (-1, 0.98)
+        self.cco.config.cbpAzAltOffset = (33.2*degrees, -2.67*degrees)
+        self.cco.config.telAzAltScale = (1, 0.95)
+        self.cco.config.telAzAltOffset = (-31.5*degrees, 2.3*degrees)
+        self.cco.config.telRotScale = -1
+        self.cco.config.telRotOffset = 222.2*degrees
+        for cbpAzAltObserved in (
+            SpherePoint(1, 2, degrees),
+            SpherePoint(-45, 75.2, degrees),
+        ):
+            self.cco.cbpAzAltObserved = cbpAzAltObserved
+            self.assertSpherePointsAlmostEqual(self.cco.cbpAzAltObserved, cbpAzAltObserved)
```

Comment:
Is this literally just testing the setter? Fine if so, just wanted to make sure this is doing what it's supposed to.

Reply:
Yes, exactly.